File: front/src/fieldReportProjectActivity/model/FieldReportProjectActivity.js
class FieldReportProjectActivities extends Array {}

const fieldReportProjectActivity_api_adapter = fieldReportProjectActivity => {
    return fieldReportProjectActivity;
};

const fieldReportProjectActivity_view_adapter = fieldReportProjectActivity => {
    return fieldReportProjectActivity;
};

const fieldReportProjectActivities_api_adapter = fieldReportProjectActivities => {
    return fieldReportProjectActivities.map(fieldReportProjectActivity_api_adapter);
};

const createFieldReportProjectActivities = (data = []) => {
    const fieldReportProjectActivities = FieldReportProjectActivities.from(
        data,
        fieldReportProjectActivity =>
            createFieldReportProjectActivity(fieldReportProjectActivity)
    );
    return fieldReportProjectActivities;
};

const createFieldReportProjectActivity = ({
    id = null,
    title = "",
    date = null,
    notes = "",
    image1 = null,
    image2 = null,
    image3 = null,
    image4 = null,
    images = [],
    folder = "",
    created_by = "",
    created_at = null,
    updated_at = null,
    updated_by = "",
} = {}) => {
    const publicApi = {
        id,
        title,
        date,
        notes,
        image1,
        image2,
        image3,
        image4,
        images,
        folder,
        created_by,
        created_at,
        updated_at,
        updated_by,
    };

    return Object.freeze(publicApi);
};

export {
    createFieldReportProjectActivity as default,
    createFieldReportProjectActivities,
    fieldReportProjectActivity_api_adapter,
    fieldReportProjectActivities_api_adapter,
    fieldReportProjectActivity_view_adapter,
};

File: back/app/models/field_report_project_activity.py
from django.db import models

from app.base.models.base_models import BaseEntityModelMixin
from app.models.field_report_project import FieldReportProject
from documents.base.base_models import BaseDocumentModel
from documents.models import Folder, MediaNode


class FieldReportProjectActivity(BaseDocumentModel, BaseEntityModelMixin):
    class Meta(object):
        db_table = "field_report_project_activity"
        verbose_name = "Actividad de proyecto en informe de viaje"
        verbose_name_plural = "Actividades de proyecto en informe de viaje"

    title = models.TextField("Título")
    date = models.DateField("Fecha del informe")
    notes = models.TextField("Notas", null=True)
    field_report_project = models.ForeignKey(
        FieldReportProject,
        on_delete=models.CASCADE,
        verbose_name=FieldReportProject._meta.verbose_name,
        null=True,
        related_name="field_report_project_activities",
    )

    image1 = models.ForeignKey(
        MediaNode,
        on_delete=models.SET_NULL,
        verbose_name="Imagen 1",
        null=True,
        related_name="image1+",
    )
    image2 = models.ForeignKey(
        MediaNode,
        on_delete=models.SET_NULL,
        verbose_name="Imagen 2",
        null=True,
        related_name="image2+",
    )
    image3 = models.ForeignKey(
        MediaNode,
        on_delete=models.SET_NULL,
        verbose_name="Imagen 3",
        null=True,
        related_name="image3+",
    )
    image4 = models.ForeignKey(
        MediaNode,
        on_delete=models.SET_NULL,
        verbose_name="Imagen 4",
        null=True,
        related_name="image4+",
    )

    def post_create(self, sender, created, *args, **kwargs):
        if created:
            folder = Folder(
                media_type="FOLDER",
                media_name="FieldReportProjectActivity_{0}".format(self.id),
                storage_path="{0}/fieldreportprojectactivity/{1}".format(
                    self.field_report_project.folder.storage_path, self.id
                ),
                parent=self.field_report_project.folder,
                creation_user=self.created_by,
            )
            folder.save()

            self.folder = folder
            self.save()
